Extension/src/LanguageServer/configurations.ts

```diff
@@ -739,29 +740,50 @@ export class CppProperties {
         const result: string[] = [];
         if (paths) {
             paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
-            return result;
+            if (paths && !glob) {
+                paths.forEach(entry => {
+                    const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+                    result.push(...entries);
+                });
+                return result;
+            }
         }
 
         const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
-            }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
-            // fastGlob silently strip non-found paths. limit that behavior to dynamic paths only
-            const matches: string[] = fastGlob.isDynamicPattern(normalized) ?
-                fastGlob.sync(normalized, { onlyDirectories: true, cwd}) : [res];
-            globResult.push(...matches.map(s => recursive ? s + '**' : s));
+        if (paths) {
```

Comment:
The path checks at line 741 and 753 can be turned into one at the beginning of the function:

example:

```ts
// If path is not defined return result immediately.
const result: string[] = []
if (paths === undefined) {
    return result;
}
// Do work for when path has value.
```